guides/common/modules/proc_creating-a-local-file-repository.adoc

```diff
@@ -12,38 +12,65 @@ To create a file type repository in a directory on a remote server, see xref:Cre
 
 To create a file type repository in a local directory, complete the following procedure:
 
-. Ensure the Server and {project-client-name} repositories are enabled.
 ifdef::satellite[]
+. Ensure the Server and {project-client-name} repositories are enabled.
+** For {EL} 8:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# subscription-manager repos --enable={RepoRHEL8BaseOS} \
+--enable={RepoRHEL8AppStream} \
+--enable={RepoRHEL8ServerSatelliteUtils}
+----
+** For {EL} 7:
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
 # subscription-manager repos --enable={RepoRHEL7Server} \
---enable={project-client-RHEL7-url}
+--enable={RepoRHEL7ServerSatelliteUtils}
 ----
 endif::[]
 . Install the Pulp Manifest package:
+** For {EL} 8:
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
 # {package-install-project} python3-pulp_manifest
 ----
+ifdef::satellite[]
 +
 Note that this command stops the {Project} service and re-runs {foreman-installer}.
 Alternatively, to prevent downtime caused by stopping the service, you can use the following:
 +
 [options="nowrap" subs="+quotes,attributes"]
 ----
-ifdef::satellite[]
-# subscription-manager repos --enable {RepoRHEL7ServerSatelliteToolsProductVersion}
-endif::[]
 # {foreman-maintain} packages unlock
-# yum install install python-pulp-manifest -y
+# {package-install} python3-pulp_manifest
 # {foreman-maintain} packages lock
-ifdef::satellite[]
-# subscription-manager repos --disable {RepoRHEL7ServerSatelliteToolsProductVersion}
+----
 endif::[]
++
+This installs the package without downtime.
+** For {EL} 7:
```

Comment:
Have you thought about the follow up? Will you submit a PR after this to drop EL7 instructions on current branches? I mostly ask because it's broken now on Katello where it tells users to run `dnf install` which doesn't work on EL7. On the other hand, we can't hardcode `yum install` since that breaks the Satellite case.

Reply:
> Have you thought about the follow up? Will you submit a PR after this to drop EL7 instructions on current branches? I mostly ask because it's broken now on Katello where it tells users to run `dnf install` which doesn't work on EL7. On the other hand, we can't hardcode `yum install` since that breaks the Satellite case.

Then we should drop instructions for RHEL 7 in both the steps, correct? which we added as a part of this PR...

@lpramuk could you please confirm if we can get rid of the RHEL 7 instructions as suggested here?

Reply:
It depends. You specified you want to cherry pick this to 3.1, where it is supported. See https://github.com/theforeman/foreman-documentation/milestone/2 for the whole table. I think you will need to write a patch for every branch with various `ifdef` statements. At least for `master` documenting EL7 makes no sense since no platform supports it anymore. It can make sense to first submit a change to document it, cherry pick that and then drop it again but it may be faster to only document EL8 in master, cherry pick that to 3.4. Then submit a different PR to 3.3 and yet another to 3.1 (Foreman 3.2 is EOL so you don't have to do that PR).

Reply:
> It depends. You specified you want to cherry pick this to 3.1, where it is supported. See https://github.com/theforeman/foreman-documentation/milestone/2 for the whole table. I think you will need to write a patch for every branch with various `ifdef` statements. At least for `master` documenting EL7 makes no sense since no platform supports it anymore. It can make sense to first submit a change to document it, cherry pick that and then drop it again but it may be faster to only document EL8 in master, cherry pick that to 3.4. Then submit a different PR to 3.3 and yet another to 3.1 (Foreman 3.2 is EOL so you don't have to do that PR).

IMO, if we are going to remove RHEL 7 support over the time, we should refrain from adding new such references. But I would wait for @lpramuk to revert on this.
Usually we keep docs for the latest release supported. In this case, RHEL 8 info should be preserved only.

Reply:
@dubewarsagar You can't drop EL7 as you state you want to cherrypick to 3.4, 3.3, 3.2 ,3.1
And EL8 only versions are 3.5 (nightly), 3.4  

RE yum VS. dnf
you differentiate in rpm name already so you can change also command for el8 to `dnf` for el7 to `yum`

But generally it is better to refer always to yum as it work on both EL7 and EL8 (where yum is symlinked to dnf)
RHEL8 documentation does so (uses yum exclusively)
I get this is upstream so many strive to have `dnf` for EL8 - but then you are required to split all steps for each EL

Reply:
@dubewarsagar I would proceed with keeping EL7 in order to cherrypick to older versions (`3.3`, `3.1`) and then you can do EL7 removal for `3.5 nightly` and `3.4`

Reply:
> @dubewarsagar I would proceed with keeping EL7 in order to cherrypick to older versions (`3.3`, `3.1`) and then you can do EL7 removal in another PR for `3.5 nightly` and `3.4`

Sounds good to me. 
@ekohl should we merge this now and then I will create another PR for EL7 removal in another PR for `3.5 nightly` and `3.4`?

Reply:
Given how much they differ I'd personally open a new PR for every branch. There have been too many changes around macros etc so I'm not going to blindly cherry pick this. A rendered preview is really useful. If you're going to do that, you might as well do it right the first time.

Note that you can still reuse this work by using `git cherry-pick` manually. For example:
```shell
# assuming origin is your remote to this repo
git checkout -b local-repo-for-3.3 origin/3.3-stable
git cherry-pick BZ-2123940-Custom-File-Type-Repo-Utils
```

The biggest reason I'd go for the correct instructions the first time is that right after merging it will be live. That means users can see invalid instructions. I really prefer to avoid that and get into a pattern where we only merge work that's ready for consumption.

Reply:
Created a new https://github.com/theforeman/foreman-documentation/pull/1671 for foreman 3.1 and 3.3. 
Once #1671 gets merged will update information in this PR#1650 to remove the EL 7 related instructions to keep it specifically for 3.4 and above. 

@ekohl pls take a look and see if the cherry-picking versions are selected correctly in #1671.

Thanks!

Reply:
I am not following the logic for EL7 removal. This is about the Satellite Utils repository which exists and will exist for RHEL 7 and RHEL 8. Therefore I see no need to swing through and remove the EL7 part. While the server may drop EL7 support, the intent of the Utils repository is to support more OS footprints that users may want or need to install tooling on.

Reply:
Maybe @ekohl can comment on this.

Reply:
I mostly suggested it because of these two lines (from line 8):
> Use this procedure to configure a repository in a directory on the base system where {Project} is installed.
> To create a file type repository in a directory on a remote server, see xref:Creating_a_Remote_File_Type_Repository_{context}[].

Since we can only install the base system on EL8, documenting EL7 is only going to add confusion.

Reply:
so are you saying we should retain EL 7 information in all the versions? 
If yes, we can merge this into master right away and there is no need of PR#1671.

Reply:
I'm saying that in this file (local installation) we only need EL8. In the remote case we can still document EL7.

Reply:
> I'm saying that in this file (local installation) we only need EL8. In the remote case we can still document EL7.

Got it. Removed EL 7 related information this module. Copying the same changes in PR#1671.